### ISSUE TITLE
[Monitoring v2] Cannot add Webhook Receiver

### ISSUE THREAD

**Issue by mscbpi** [rancher/rancher#30421]
Rancher 2.5.3

Trying to add a Webhook receiver in the monitoring module.

`Cannot read property 'data' of undefined`

```
Uncaught (in promise) TypeError: Cannot read property 'data' of undefined
    at k (alertmanagerconfig.js:38)
    at alertmanagerconfig.js:54
    at d (runtime.js:45)
    at Generator._invoke (runtime.js:274)
    at Generator.O.forEach.t.<computed> [as next] (runtime.js:97)
    at r (asyncToGenerator.js:3)
    at l (asyncToGenerator.js:25)
```

```
function extractConfig(secret) {
  secret.data = secret.data || {};
  const file = secret.data[FILENAME];
  const decodedFile = file ? base64Decode(file) : '{}';
  const config = jsyaml.safeLoad(decodedFile);

  config.receivers = config.receivers || [];
  config.route = config.route || {};
  config.route.routes = config.route.routes || [];
```

**Comment by mscbpi** [comment]
Actually happens with whatever receiver we try to configure

**Comment by sweetslobber** [comment]
+1 we have the same problems

**Comment by mscbpi** [comment]
Uninstall and reinstall the monitoring. For some reasons it has not parsed the config that's in the secret. It comes back.

**Comment by rondlite** [comment]
> Uninstall and reinstall the monitoring. For some reasons it has not parsed the config that's in the secret. It comes back.

tried that, twice.. still can not configure webhook.. what config should have which value in it?

**Comment by rondlite** [comment]
I succeeded after also purging the alertmanager secrets and reinstalling. it has something to do with the `null` that is made by default..

**Comment by mscbpi** [comment]
the `null` reflects the alertmanager default receiver in the config (check the config in the `secret`). It is expected.

**Comment by rondlite** [comment]
> the `null` reflects the alertmanager default receiver in the config (check the config in the `secret`). It is expected.

I think for me the trouble started because I deleted it from the list and wanted to create alert after. 
Now just leaving the null there as is and is all good. 
Might be idea to make the null record undeletable.

**Comment by mscbpi** [comment]
Edit your rules with the yaml these are standardized rules. You'll understand it better. At the moment the GUI may not be consistent in interpreting and managing the config.

**Comment by jiaqiluo** [comment]
Hi @mscbpi @sweetslobber  @rondlite  Thank you for reporting the bug. 

This is a known [UI issue](https://github.com/rancher/dashboard/issues/1845) and the fix will be in the coming rancher:`v2.5.4` release.

**Comment by mscbpi** [comment]
Great, it is indeed corrected.
